src/Auth0.AuthenticationApi/AuthenticationApiClient.cs

```diff
@@ -285,6 +285,32 @@ public Task<AccessTokenResponse> GetTokenAsync(PasswordlessSmsTokenRequest reque
                 body);
         }
 
+        /// <inheritdoc/>
+        public async Task<AccessTokenResponse> GetTokenAsync(DeviceCodeTokenRequest request)
+        {
+            if (request == null) throw new ArgumentNullException(nameof(request));
+
+            var body = new Dictionary<string, string>
+            {
+                {"grant_type", "urn:ietf:params:oauth:grant-type:device_code"},
+                {"device_code", request.DeviceCode},
+                {"client_id", request.ClientId}
+            };
+
+            // TODO: This will throw an exception while the user is completing their part of the flow. From what
+            // I remember the "error" is recoverable via ErrorApiException.ApiError.Error but not "error_description"
+            var response = await connection.SendAsync<AccessTokenResponse>(
+                HttpMethod.Post,
+                tokenUri,
+                body
+            ).ConfigureAwait(false);
+
+            // TODO: Validate tokens - this is recommended but I haven't implemented it in my spike
+            // await AssertIdTokenValid(response.IdToken, request.ClientId, request.SigningAlgorithm, request.ClientSecret).ConfigureAwait(false);
```

Comment:
Let's add this in and remove the comments.

Reply:
Looking in to this in a bit more depth, I don't think the `AssertIdTokenValid` is possible since this flow doesn't have a `ClientSecret`. I notice that the 2 password exchanges above this method don't validate the token either.

Obviously happy to defer to your better knowledge, but I would need some guidance.

Reply:
ClientSecret is only used in case of HS256, not when using RS256, which is the recommend Signature Algorithm anyway.
For now, I would only support RS256 in this endpoint, avoiding the need for a ClientSecret.

`await AssertIdTokenValid(response.IdToken, request.ClientId, JwtSignatureAlgorithm.RS256, null).ConfigureAwait(false);`